src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -34698,32 +34629,154 @@ .locals init (System.Collections.Generic.List<T> V_0,
+        public void List_SingleSpread_IEnumerable()
         {
             var source = """
-                using System.Collections;
                 using System.Collections.Generic;
 
-                class MyCollection(List<int> list) : IEnumerable<int>
+                class C
                 {
-                    public IEnumerator<int> GetEnumerator() => list.GetEnumerator();
+                    static void Main()
+                    {
+                        M([1, 2, 3]).Report();
+                    }
 
-                    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                    static List<int> M(IEnumerable<int> e) => [..e];
+                }
+                """;
+
+            var verifier = CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3],", verify: Verification.Skipped);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("C.M", """
+                {
+                  // Code size        7 (0x7)
+                  .maxstack  1
+                  IL_0000:  ldarg.0
+                  IL_0001:  call       "System.Collections.Generic.List<int> System.Linq.Enumerable.ToList<int>(System.Collections.Generic.IEnumerable<int>)"
+                  IL_0006:  ret
                 }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71217")]
+        public void List_SingleSpread_IEnumerable_Covariant()
+        {
+            var source = """
+                using System.Collections.Generic;
 
                 class C
                 {
                     static void Main()
                     {
-                        M(new([1, 2, 3])).Report();
+                        M(["a", "b", "c"]).Report();
                     }
 
-                    static List<int> M(MyCollection c) => [..c];
+                    static List<object> M(IEnumerable<string> e) => [..e];
                 }
                 """;
 
-            var verifier = CompileAndVerify([source, s_collectionExtensions], expectedOutput: IncludeExpectedOutput("[1, 2, 3],"), targetFramework: TargetFramework.Net80, verify: Verification.Skipped);
+            var verifier = CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[a, b, c],", verify: Verification.Skipped);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("C.M", """
+                {
+                  // Code size        7 (0x7)
+                  .maxstack  1
+                  IL_0000:  ldarg.0
+                  IL_0001:  call       "System.Collections.Generic.List<object> System.Linq.Enumerable.ToList<object>(System.Collections.Generic.IEnumerable<object>)"
+                  IL_0006:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/71217")]
+        public void List_SingleSpread_IEnumerable_ClassConstraint()
```

Comment:
The unconstrained and `struct`-constrained cases are considered as boxing conversions, therefore they are verified by `List_AddRange_IEnumerable_Constraint` test